### ISSUE TITLE
Fix false positives for `#RRGGBBAA` in `color-named`

### ISSUE THREAD

**Issue by ota-meshi** [stylelint/stylelint#5716]
### What steps are needed to reproduce the bug?

```css
.a { color: #fff0ffff; }
.b { color: azure; }
```

[DEMO](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0BDABMNkA2EBO8aAxAGZkAMZZA3GgL4A6AdkgEYZYS4FooBeAVzwxadEABoQAZwCeTAC4oAHgnBSpE8BCYkAlgHNVwZmjQMQeAdhhTzhY01OnzOfAFomKALYwod8yjYAO4oMlJuQQAWMMJuAA4Q6rqs1uYm9MxidEA)

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "color-named":"always-where-possible"
  }
}
```

### How did you run Stylelint?

cli

### Which version of Stylelint are you using?

14.0.1

### What did you expect to happen?

No problems to be reported.

### What actually happened?

The `color-named` rule warns you to replace `#fff0ffff` with `azure`, but they are completely different colors.

![image](https://user-images.githubusercontent.com/16508807/141427662-42346b70-7d44-4012-ae13-52bc5307bbd6.png)

```
Expected "#fff0ffff" to be "azure" (color-named)
```

**Comment by jeddy3** [comment]
@ota-meshi Thanks for the report. 

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.
